crates/bindings-csharp/BSATN.Runtime/Db.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace SpacetimeDB;
+
+public abstract record DbContext<DbView>
+    where DbView : class, new()
+{
+    public readonly DbView Db;
+
+    public DbContext() => Db = new();
+
+    public DbContext(DbView db) => Db = db;
+}
```

Comment:
I meant you can then use the shorthand syntax, not just change class to struct.

So just:

```suggestion
public abstract record DbContext<DbView>(DbView Db)
    where DbView : class, new()
{
    // Do we really need this constructor btw?
    public DbContext() : this(new()) { }
}
```

Reply:
Ah I keep forgetting that syntax exists hah, thanks!
The empty ctor is needed for derived classes that don't specify one.

Reply:
> The empty ctor is needed for derived classes that don't specify one.

It seems there is only one such class (`BaseReducerContext`) - let's move the creation of `DbView` to it instead? Then it will be just `record BaseReducerContext<DbView> : DbContext<DbView>(new())`.